GroversAlgorithm/ReferenceImplementation.qs

```diff
@@ -36,19 +36,22 @@ namespace Quantum.Kata.GroversAlgorithm
         {
             // flip the bits in odd (0-based positions), 
             // so that the condition for flipping the state of the target qubit is "query register is in 1...1 state"
-            for (n in 0..Length(queryRegister)-1)
-            {
-                if (n % 2 == 1)
-                {
-                    X(queryRegister[n]);
-                }
-            }
+            FlipOddPositionBits_Reference(queryRegister);
             (Controlled X)(queryRegister, target);
-            for (n in 0..Length(queryRegister)-1)
+            FlipOddPositionBits_Reference(queryRegister);
```

Comment:
This should actually be 
```suggestion
            (Adjoint FlipOddPositionBits_Reference)(queryRegister);
```
(and the operation should be declared as having an adjoint), since semantically the second bitflip is duplicate code but uncomputing the effect of the previous bitflip